src/csharp/Microsoft.Spark.UnitTest/ConfigurationServiceTests.cs

```diff
@@ -0,0 +1,184 @@
+                Assert.Equal(
+                    _workerDirEnvVars.WorkerMajorMinorBuildDir.Name, configService.WorkerDirEnvVarName);
+            }
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithNoEnvVars()
+        {
+            var configService = new ConfigurationService();
+
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerDir.IsSet());
+
+            // Environment variables not set, only Microsoft.Spark.Worker filename should be returned.
+            Assert.Equal(ConfigurationService.ProcFileName, configService.GetWorkerExePath());
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithWorkerDirEnvVar()
+        {
+            var configService = new ConfigurationService();
+            var workerDir = "workerDir";
+            Environment.SetEnvironmentVariable(_workerDirEnvVars.WorkerDir.Name, workerDir);
+
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorDir.IsSet());
+            Assert.True(_workerDirEnvVars.WorkerDir.IsSet());
+
+            // Only WorkerDir is set, WorkerExePath will be built using it.
+            Assert.Equal(
+                Path.Combine(workerDir, ConfigurationService.ProcFileName),
+                configService.GetWorkerExePath());
+        }
+
+        [Fact]
+        public void TestWorkerExePathWithEnvVarPrecedence()
+        {
+            var configService = new ConfigurationService();
+            var workerDir = "workerDir";
+            Environment.SetEnvironmentVariable(_workerDirEnvVars.WorkerDir.Name, workerDir);
+            var workerMajorDir = "workerMajorDir";
+            Environment.SetEnvironmentVariable(_workerDirEnvVars.WorkerMajorDir.Name, workerMajorDir);
+
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorBuildDir.IsSet());
+            Assert.False(_workerDirEnvVars.WorkerMajorMinorDir.IsSet());
+            Assert.True(_workerDirEnvVars.WorkerMajorDir.IsSet());
+            Assert.True(_workerDirEnvVars.WorkerDir.IsSet());
+
+            // WorkerDir and WorkerMajorDir environment variables are set.
+            // Ensure that the environment variable constructed with the
+            // assembly version takes precedence.
+            Assert.Equal(
+                Path.Combine(workerMajorDir, ConfigurationService.ProcFileName),
+                configService.GetWorkerExePath());
+        }
+
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable(
+                _workerDirEnvVars.WorkerDir.Name,
+                _workerDirEnvVars.WorkerDir.Value);
+            Environment.SetEnvironmentVariable(
```

Comment:
Don't we need to unset others and just leave the default one (we should save the previous value in the constructor and use it?), if I understand this correctly?

Reply:
I'm saving all previously set env var names and values in the constructor, and then unsetting all the values.  In the tests I'm setting the env vars, doing validation, and then in dispose the original env var values will be restored.